dnn_test.go

```diff
@@ -283,6 +283,81 @@ func TestTensorflowMemory(t *testing.T) {
 	checkTensorflowNet(t, net)
 }
 
+func TestOnnxMemory(t *testing.T) {
+	path := os.Getenv("GOCV_ONNX_TEST_FILES")
+	if path == "" {
+		t.Skip("Unable to locate ONNX model file for tests")
+	}
+
+	b, err := ioutil.ReadFile(path + "/googlenet-9.onnx")
+	if err != nil {
+		t.Errorf("Failed to load ONNX from file: %v", err)
+	}
+
+	net, err := ReadNetFromONNXBytes(b)
+	if err != nil {
+		t.Errorf("Failed to load Tensorflow model from bytes: %v", err)
+	}
+	if net.Empty() {
+		t.Errorf("Unable to load Tensorflow model")
+	}
+	defer net.Close()
+
+	checkONNXNet(t, net)
+}
+
+func TestOnnxDisk(t *testing.T) {
+	path := os.Getenv("GOCV_ONNX_TEST_FILES")
+	if path == "" {
+		t.Skip("Unable to locate ONNX model file for tests")
+	}
+
+	net := ReadNetFromONNX(path + "/googlenet-9.onnx")
+	if net.Empty() {
+		t.Errorf("Unable to load ONNX model")
+	}
+	defer net.Close()
+
+	checkONNXNet(t, net)
+}
+
+func checkONNXNet(t *testing.T, net Net) {
+	img := IMRead("images/space_shuttle.jpg", IMReadColor)
+	if img.Empty() {
+		t.Error("Invalid Mat in ONNX test")
+	}
+	defer img.Close()
+
+	blob := BlobFromImage(img, 1.0, image.Pt(224, 224), NewScalar(0, 0, 0, 0), true, false)
+	if blob.Empty() {
+		t.Error("Invalid blob in ONNX test")
+	}
+	defer blob.Close()
+
+	net.SetInput(blob, "data_0")
```

Comment:
are "data_0" and :prob_1" names of the specific network you're testing?

Reply:
Yes, it's the input and output names specific to this model